src/app/shared/service/confirm_equal_validator.directive.ts

```diff
@@ -0,0 +1,22 @@
+import { AbstractControl } from '@angular/forms';
```

Comment:
Please avoid  " _" character while naming files. Please follow this style guide
https://angular.io/guide/styleguide